### PR TITLE
txn: fix point-lock non-existent key with aggressive locking under rc

### LINKED ISSUE #41527: Enabling aggressive locking makes TestPointGetLockExistKey fail
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

`set @@tidb_pessimistic_txn_aggerssive_locking=1` in test `TestPointGetLockExistKey`

### 2. What did you expect to see? (Required)

test pass.

### 3. What did you see instead (Required)

error: pessimistic lock not found for RC isolation level.

### 4. What is your TiDB version? (Required)

master (v6.6.0)

### PR DESCRIPTION
**Pull request by ekexium**
### What problem does this PR solve?

Issue Number: close #41527 

Problem Summary:

Under RC, point-get with lock on a non-existent key will not actually lock it. Aggressive locking didn't handle this case well and caused the committer to prewrite this non-existent key, thus generating "pessimistic lock not found" error.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -274,5 +274,6 @@ replace (
 	// fix potential security issue(CVE-2020-26160) introduced by indirect dependency.
 	github.com/dgrijalva/jwt-go => github.com/form3tech-oss/jwt-go v3.2.6-0.20210809144907-32ab6a8243d7+incompatible
 	github.com/pingcap/tidb/parser => ./parser
+	github.com/tikv/client-go/v2 => github.com/ekexium/client-go/v2 v2.0.0-alpha.0.20230216162745-b5208457bdac
```
Comment (zyguan) [inline go.mod:277]: We can switch back to the latest client-go since https://github.com/tikv/client-go/pull/705 has been merged.

**Review by zyguan — APPROVED** [review]
Rest LGTM.

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**